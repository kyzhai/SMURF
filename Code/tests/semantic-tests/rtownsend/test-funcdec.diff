11c11
< 		f x m :: Unknown = 3 + x / m
---
> 		f (x) (m) :: Int -> Int -> Int = 3 + x / m
21d20
< 	ID: main  :: Unknown
22a22
> 	ID: main  :: Unknown
24a25,26
> 	ID: head h : t :: [a] -> a
> 	ID: tail h : t :: [a] -> [a]
