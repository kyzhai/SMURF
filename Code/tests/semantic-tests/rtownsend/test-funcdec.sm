f :: Int -> Int -> Int
x = 5
m = x + 2
f x m = 3 + x / m

main = []
