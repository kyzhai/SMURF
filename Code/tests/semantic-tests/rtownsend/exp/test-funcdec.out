Program: 
	SVardef: 
		ID: x  :: Int

	5
	SVardef: 
		ID: m  :: Int

	x + 2
	SFuncdec: 
		f (x) (m) :: Int -> Int -> Int = 3 + x / m
	New Scope: 
		ID: x  :: Int
		ID: m  :: Int

	
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: x  :: Int
	ID: m  :: Int
	ID: f x m :: Int -> Int -> Int
	ID: print x :: a -> a
	ID: random  :: Int

